vllm/prompt_adapter/models.py

```diff
@@ -0,0 +1,279 @@
+import logging
+import math
+from typing import Callable, Dict, List, Optional, Type
+
+from peft.utils import load_peft_weights
+from torch import nn
+
+from vllm.adapter_commons.models import (AdapterLRUCache, AdapterModel,
+                                         AdapterModelManager)
+from vllm.config import PromptAdapterConfig
+from vllm.prompt_adapter.layers import (
+    PromptAdapterMapping, VocabParallelEmbeddingWithPromptAdapter)
+
+logger = logging.getLogger(__name__)
+
+_GLOBAL_PROMPT_ADAPTER_ID = 0
+
+
+def get_prompt_adapter_id():
+    global _GLOBAL_PROMPT_ADAPTER_ID
+    _GLOBAL_PROMPT_ADAPTER_ID += 1
+    return _GLOBAL_PROMPT_ADAPTER_ID
+
+
+class PromptAdapterModel(AdapterModel):
+
+    def __init__(self,
+                 prompt_adapter_id=None,
+                 num_virtual_tokens=None,
+                 prompt_embedding=None) -> None:
+        self.id = prompt_adapter_id
+        self.kv_cache = None
+        self.prompt_embedding = prompt_embedding
+        self.num_virtual_tokens = num_virtual_tokens
+
+    @classmethod
+    def from_local_checkpoint(cls,
+                              adapter_model_and_path,
+                              prompt_adapter_id,
+                              torch_device='cuda') -> "PromptAdapterModel":
+        adapters_weights = load_peft_weights(adapter_model_and_path,
+                                             torch_device)
+        prompt_embedding = adapters_weights["prompt_embeddings"].half()
```

Comment:
do we always want to cast to float16? this should probably be the actual model dtype.

Reply:
sounds good

Reply:
this has been changed to use the model dtype